pkg/kwokctl/cmd/export/logs/logs.go

```diff
@@ -0,0 +1,158 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package logs implements the `logs` command
+package logs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/kwok/pkg/config"
+	"sigs.k8s.io/kwok/pkg/kwokctl/runtime"
+	"sigs.k8s.io/kwok/pkg/log"
+	"sigs.k8s.io/kwok/pkg/utils/concurrent"
+	"sigs.k8s.io/kwok/pkg/utils/file"
+	"sigs.k8s.io/kwok/pkg/utils/version"
+)
+
+type flagpole struct {
+	Name string
+}
+
+// NewCommand returns a new cobra.Command for getting the cluster logs
+func NewCommand(ctx context.Context) *cobra.Command {
+	flags := &flagpole{}
+	cmd := &cobra.Command{
+		Args: cobra.MaximumNArgs(1),
+		// TODO(bentheelder): more detailed usage
+		Use:   "logs [output-dir]",
+		Short: "Exports logs to a tempdir or [output-dir] if specified",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			flags.Name = config.DefaultCluster
+			return runE(ctx, flags, args)
+		},
+	}
+	return cmd
+}
+
+func runE(ctx context.Context, flags *flagpole, args []string) error {
+	name := config.ClusterName(flags.Name)
+	workdir := path.Join(config.ClustersDir, flags.Name)
+
+	logger := log.FromContext(ctx).With("cluster", flags.Name)
+	ctx = log.NewContext(ctx, logger)
+
+	rt, err := runtime.DefaultRegistry.Load(ctx, name, workdir)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			logger.Warn("Cluster is not exists")
+		}
+		return err
+	}
+
+	// get the optional directory argument, or create a tempdir
+	var dir string
+	if len(args) == 0 {
+		t, err := file.TempDir("", "")
+		if err != nil {
+			return err
+		}
+		dir = t
+	} else {
+		dir = args[0]
+	}
+
+	logger.Info("Exporting logs to ", dir)
+
+	// ensure directory
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+		return fmt.Errorf("failed to create logs directory")
+	}
+
+	if err := os.WriteFile(
+		filepath.Join(dir, "kwok-version.txt"),
+		[]byte(version.DisplayVersion()),
+		0640,
+	); err != nil {
+		return fmt.Errorf("failed to write kwok-version.txt %w", err)
+	}
+
+	cfg, err := rt.Config(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = config.Save(ctx, filepath.Join(dir, "kwok.yaml"), []config.InternalObject{cfg})
+	if err != nil {
+		return err
+	}
+
+	logToPathFn := func(path, name string) func() error {
+		return func() error {
+			f, err := file.OnHost(path)
+			if err != nil {
+				return err
+			}
+			defer func() {
+				_ = f.Close()
+				if err != nil {
+					_ = os.Remove(path)
+				}
+			}()
+			return rt.Logs(ctx, name, f)
+		}
+	}
+
+	// construct a slice of methods to collect logs
+	fns := make([]func() error, 0, len(cfg.Components)+1)
+
+	for _, component := range cfg.Components {
+		fns = append(fns, logToPathFn(filepath.Join(filepath.Join(dir, name), component.Name+".log"), component.Name))
```

Comment:
Use `rt.Logs` instead of get from file and this logic moves into `rt.CollectLogs`